tests/test_browser.go

```diff
@@ -0,0 +1,297 @@
+func (b *browser) withHandle(pattern string, handler http.HandlerFunc) *browser {
+	if b.http == nil {
+		panic("You should enable HTTP test server, see: withHTTPServer option")
+	}
+	b.http.Mux.Handle(pattern, handler)
+	return b
+}
+
+const testBrowserStaticDir = "static"
+
+// withFileServer serves a file server using the HTTP test server that is
+// accessible via `testBrowserStaticDir` prefix.
+//
+// This method is for enabling the static file server after starting a test
+// browser. For early starting the file server see withFileServer function.
+func (b *browser) withFileServer() *browser {
+	const (
+		slash = string(os.PathSeparator)
+		path  = slash + testBrowserStaticDir + slash
+	)
+
+	fs := http.FileServer(http.Dir(testBrowserStaticDir))
+
+	return b.withHandle(path, http.StripPrefix(path, fs).ServeHTTP)
+}
+
+// URL returns the listening HTTP test server's URL combined with the given path.
+func (b *browser) URL(path string) string {
+	if b.http == nil {
+		panic("You should enable HTTP test server, see: withHTTPServer option")
+	}
+	return b.http.ServerHTTP.URL + path
+}
+
+// staticURL is a helper for URL("/`testBrowserStaticDir`/"+ path).
+func (b *browser) staticURL(path string) string {
+	return b.URL("/" + testBrowserStaticDir + "/" + path)
+}
+
+// attachFrame attaches the frame to the page and returns it.
+func (b *browser) attachFrame(page api.Page, frameID string, url string) api.Frame {
+	pageFn := `
+	async (frameId, url) => {
+		const frame = document.createElement('iframe');
+		frame.src = url;
+		frame.id = frameId;
+		document.body.appendChild(frame);
+		await new Promise(x => frame.onload = x);
+		return frame;
+	}
+	`
+	return page.EvaluateHandle(
+		b.rt.ToValue(pageFn),
+		b.rt.ToValue(frameID),
+		b.rt.ToValue(url)).
+		AsElement().
+		ContentFrame()
+}
+
+// detachFrame detaches the frame from the page.
+// func (b *browser) detachFrame(page api.Page, frameID string) {
+// 	pageFn := `
+// 	frameId => {
+//         	document.getElementById(frameId).remove();
+//     	}
+// 	`
+// 	page.Evaluate(
+// 		b.rt.ToValue(pageFn),
+// 		b.rt.ToValue(frameID))
+// }
+
+// launchOptions provides a way to customize browser type
+// launch options in tests.
+type launchOptions struct {
+	Debug    bool   `js:"debug"`
+	Headless bool   `js:"headless"`
+	SlowMo   string `js:"slowMo"`
+	Timeout  string `js:"timeout"`
+}
+
+// withLaunchOptions is a helper for increasing readability
+// in tests while customizing the browser type launch options.
+//
+// example:
+//
+//    b := TestBrowser(t, withLaunchOptions{
+//        SlowMo:  "100s",
+//        Timeout: "30s",
+//    })
+type withLaunchOptions = launchOptions
+
+// defaultLaunchOptions returns defaults for browser type launch options.
+// TestBrowser uses this for launching a browser type by default.
+func defaultLaunchOpts() launchOptions {
+	var (
+		debug    = false
+		headless = true
+	)
+	if v, found := os.LookupEnv("XK6_BROWSER_TEST_DEBUG"); found {
```

Comment:
Do we need a separate env var for this? We can already use `XK6_BROWSER_LOG`.

Reply:
Yeah, I can get it out from here.

Btw, Playwright uses `DEBUG` env var for this like `DEBUG=something:somethingElse`. Maybe we should adopt the same behavior.

Reply:
Hhmm could be, though that would require us to plug that into every separate logger/component that should support it, which could be some work. I like the current flexibility of being able to specify log levels more than which component to enable `debug` for. Plus I'd prefer we use a separate env var than overloading a generic env var like `DEBUG` which could be set for other uses.

Create an issue if you want to discuss it, but let's keep `XK6_BROWSER_LOG` for now.

Reply:
OK, will do.